src/index.js

```diff
@@ -299,7 +299,21 @@ class CompressionPlugin {
           }
 
           if (this.options.deleteOriginalAssets) {
-            // eslint-disable-next-line no-param-reassign
+            if (this.options.deleteOriginalAssets === 'keep-source-map') {
+              // TODO `...` required only for webpack@4
+              const updatedAssetInfo = {
+                ...info,
+                related: { ...info.related, sourceMap: null },
+              };
```

Comment:
Let's do it better:
```
delete updatedAssetInfo.related.sourceMap;
```

Reply:
I tried this, this is not working. The sourceMap is getting added back if a value is not provided.

Reply:
Webpack I think is doing a smart merge there. 🤔